fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -496,10 +496,45 @@ struct CommitBatchContext {
 
 	void setupTraceBatch();
 
+	void findTLogGroups();
+
 private:
 	void evaluateBatchSize();
 };
 
+void CommitBatchContext::findTLogGroups() {
```

Comment:
This method seems to follow the logic in assignMutationsToStorageServers() in order to extract the set of affected tags. But, as we have seen in https://github.com/apple/foundationdb/pull/5272, this doesn't find all affected tags - we also need to extract tags based on the logic in applyMetadataToCommittedTransactions().

Reply:
>This PR adds new code to figure out the set of affected tags. It would be nice to have an assert that verifies that we are still collecting the same set of tags. Would you please add that? 

With this change, the set of tags generated from parsing the transaction *before* the resolve step may differ from *after* the resolve step. Because any tags with conflicts are not included in the final set. I can probably add logic to handle that case, though, so an assert will work.